### ISSUE TITLE
Shared Element Transitions freezes app because of ScrollView(s)

### ISSUE THREAD

**Issue by mrousavy** [wix/react-native-navigation#6607]
### Issue Description

Sometimes when using Shared Element Transitions (SETs), the screen navigation takes **REALLY LONG**. I'm not talking about some milliseconds, I've managed to reproduce half a minute of delay. 

Now I say _sometimes_, because I unfortunately found no way to a guaranteed repro. In my case, I'm experiencing the delay on a navigation where a very small icon ("Story") gets transitioned full screen. (small circular background -> full screen background, and icon + name transition)

### Demos

1. [Here](https://i.imgur.com/EV4TQ6C.mp4) is a video demo of it getting stuck completely (it took a bit more than a minute until it began with the transition)
2. [Here](https://i.imgur.com/9kHPBXv.mp4) is a video demo where I also showcase the short delay but perform a hot reload while waiting
3. [Here](https://i.imgur.com/HDmaQHf.mp4) is a video demo of it working perfectly fine if I remove the `sharedElementTransitions` from my navigation options (`animations.push`)

### Steps to Reproduce / Code Snippets / Screenshots

1. Create small view which gets used for SET
2. Give view a colored background, an icon above and text below. Give those views a `nativeID` (for SET)
3. Create SET:
	```js
	push: {
		sharedElementTransitions: [
			{
				fromId: getStorySharedElementId(storyId, 'title', 'from'),
				toId: getStorySharedElementId(storyId, 'title', 'to'),
				interpolation: 'overshoot',
				duration: STORY_SHARED_ELEMENT_TRANSITION_DURATION,
			},
			{
				fromId: getStorySharedElementId(storyId, 'icon', 'from'),
				toId: getStorySharedElementId(storyId, 'icon', 'to'),
				interpolation: 'overshoot',
				duration: STORY_SHARED_ELEMENT_TRANSITION_DURATION,
			},
			{
				fromId: getStorySharedElementId(storyId, 'background', 'from'),
				toId: getStorySharedElementId(storyId, 'background', 'to'),
				interpolation: 'overshoot',
				duration: STORY_SHARED_ELEMENT_TRANSITION_DURATION,
			},
		],
	},
	```
4. Run SET multiple times, I've managed to reproduce the bug only with a clean app state (no hot reload).

### Notes

A few notes that might be interesting for debugging:

1. When performing a hot-reload (hit Ctrl+S on any file) while the screen is at a stuck state (in the middle of the navigation process), the navigation suddenly starts and transitions successfully.
2. I'm using a snapshot build of the API
3. I'm using the `'overshoot'` interpolation mode
4. I have a very complex screen, but I've measured it's render time and that shouldn't be the problem. I've also tried to remove the Shared Element Transitions and just use `waitForRender: true` which works perfectly. See the 3rd video demo
5. I'm using reanimated 2.0.0-alpha.7
6. The problem only happens on RNN 7.0.0-snapshot2020, on 7.0.0 everything works fine
7. I've tried to remove everything in the from-view and to-view to only animate a single `<Text>` element. Because of that, I can be sure that it's not my view that renders slow, but rather a regression introduced somewhere since the 7.0.0 release

---
### Environment
* React Native Navigation version: 7.0.0-snapshot2020
* React Native version: 0.63.2
* Platform(s) (iOS, Android, or both?): iOS only
* Device info (Simulator/Device? OS version? Debug/Release?): Simulator Debug, iPhone 11 Pro Release

**Comment by mrousavy** [comment]
Found what we need to look for! It's an issue with the bottomTabs. It's because I hide them.

This works:

```js
StoryDetailsScreen.options = {
	bottomTabs: {
		visible: true,
	},
};
```

This doesn't:

```js
StoryDetailsScreen.options = {
	bottomTabs: {
		visible: false,
	},
};
```

If I understood correctly, the bottomTabs visibility prop should be deprecated, how else am I supposed to display my Screen in fullscreen?

**Comment by mrousavy** [comment]
So I've played around with my code a little bit and noticed that it has something to do with the `PressableScale` component (can also be found in the playground) - once I replaced that with a `TouchableOpacity` from react-native, everything worked fine and the animations started instantly. The weird thing is, once I started holding the button, the animation started to bug out again, seems like the state change (`onPressIn`, `onPress`, `onPressOut` or even `onLongPress`) f*cks up the Shared Element Transition -  wtf???

**Comment by mrousavy** [comment]
I've tested this further and can confirm that it has something to do with press/touch events. I've triggered the navigation on the `onPressIn` event instead of the `onPress` event, so it should navigate once I started to press the story. Instead, it only navigated once I _release_ my "finger", see: [this video demo](https://i.imgur.com/Qystdk9.mp4). As you can see in that demo, the horizontal scroll view also behaves weirdly. That's because the scroll and scroll momentum animations cannot be run at the same time as the navigation animation (Shared Element Transition). I've also noticed this with my post screen, where the animation bugs when the scrollview has still scroll momentum, see [this demo](https://i.imgur.com/BQPY2M0.mp4).

So TL;DR: The issue is caused by the scroll-view still picking up the finger event, and when I use the `onPress` event on a single story circle, the scroll view gets stuck and still thinks I'm holding my finger down at the scrollview, so the animation doesn't start until it realizes that I'm not there anymore (which appears to happen randomly).

So TL;DR of the TL;DR: ScrollView + SET = no | Button in ScrollView + SET = no, but weirder

Of course I could play around with my scrollview's `waitFor` ref props and somehow pass it through all those buttons, but I think that's a weird workaround. What do you think @guyca & @yogevbd ?

**Comment by guyca** [comment]
Closing this one as either it's already fixed or does not reproduce since switching to modals. @mrousavy thanks for all the info you've provided!
